### PR TITLE
Create nfsserver.md

### PR DESCRIPTION
**Pull request by alemorvan**
Documentation about nfsserver

#### Author checklist (Completed by original Author)
- [x] Good fit for the Rocky Linux project? Title and Author Metatags inserted ?
- [x] If applicable, steps and instructions have been tested to work
- [x] Initial self-review to fix basic typos and grammar completed

#### Rocky Documentation checklist (Completed by Rocky team) 
- [x] 1st Pass (Document is good fit for project and author checklist completed)
- [x] 2nd Pass (Technical Review - check for technical correctness) 
- [ ] 3rd Pass (Detailed Editorial Review and Peer Review)
- [ ] Final approval (Final Review)

### CODE REVIEW

File: docs/guides/file_sharing/nfsserver.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+### The `/etc/exports` file
+
+Resource shares are set up in the `/etc/exports` file. Each line in this file corresponds to an NFS share.
+
+```
+/share_name	client1(permissions) client2(permissions)
+```
+
+* **/share_name**: Absolute path of shared directory;
+* **clients** : Clients authorized to access resources ;
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:77]:
```suggestion
* **clients**: Clients authorized to access resources;
```

File: docs/guides/file_sharing/nfsserver.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+* **(permissions)** : Permissions on resources.
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:78]:
```suggestion
* **(permissions)**: Permissions on resources.
```

File: docs/guides/file_sharing/nfsserver.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+* **FQDN**: client_*.rockylinux.org: allows FQDNs starting with client_ from the rockylinux.org domain;
+* `*` for everybody.
+
+Multiple clients can be specified on the same line, separated by a space.
+
+### Permissions on resources
+
+There are two types of permissions:
+
+* `ro` : readonly;
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:93]:
```suggestion
* `ro` : read-only;
```

File: docs/guides/file_sharing/nfsserver.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+* `rw` : modification.
+
+If no right is specified, then the right applied will be read-only.
+
+By default, client user UIDs and GIDs are conserved (except for `root`).
+
+To force the use of a UID or GID other than that of the user writing the resource, specify the `anonuid=UID` and `anongid=GID` options, or give `anonymous` access to the data with the `all squash`option.
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:100]:
```suggestion
To force the use of a UID or GID other than that of the user writing the resource, specify the `anonuid=UID` and `anongid=GID` options, or give `anonymous` access to the data with the `all squash` option.
```

File: docs/guides/file_sharing/nfsserver.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+!!! warning "warning" 
+
+    There is a parameter, `no_root_squash`, which identifies the client root user as the server root user. This parameter can be dangerous for system security.
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:104]:
```suggestion
    There is a parameter, `no_root_squash`, which identifies the client root user as the server root user. This parameter can be dangerous from a system security point of view.
```

File: docs/guides/file_sharing/nfsserver.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+The `exportfs` (exported file systems) command is used to manage the table of local files shared with NFS clients.
+
+```
+exportfs [-a] [-r] [-u share_name] [-v]
+```
+
+| Options         | Description                               |
+| --------------- | ----------------------------------------- |
+| `-a`            | Enable NFS shares                         |
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:133]:
```suggestion
| `-a`            | Enables NFS shares                         |
```

File: docs/guides/file_sharing/nfsserver.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+| `-r`            | Apply shares from the `/etc/exports` file |
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:134]:
```suggestion
| `-r`            | Applies shares from the `/etc/exports` file |
```

File: docs/guides/file_sharing/nfsserver.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+| Options | Description                               |
+| ------- | ----------------------------------------- |
+| `-e`    | Displays shares on the designated server  |
+| `-a`    | Displays all current shares on the server |
+
+This command also determines whether the client workstation has authorization to mount shared resources.
+
+!!! note "note"
+
+    `showmount`sorts and deletes duplicates in the results, so it's impossible to determine whether a client has made multiple mounts of the same directory.
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:155]:
```suggestion
    `showmount` sorts and hides duplicates in the results, so it's impossible to determine whether a client has made multiple mounts of the same directory or not.
```

File: docs/guides/file_sharing/nfsserver.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+## Client configuration
+
+Access to shared resources on an NFS server is done by mount point on the client.
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:159]: ?

**Review by SergeCroise — CHANGES_REQUESTED** [review]
Hello @alemorvan and @sspencerwire ,
Please review my proposals, feel free to overwrite my changes!

File: docs/guides/file_sharing/nfsserver.md (hunk 9 above)
Comment (alemorvan) [reply docs/guides/file_sharing/nfsserver.md:159]: I don't know how to explain that in english.
It means you have to create a mount point on the client side to access to the data on server.

**Comment by alemorvan** [comment]
Thx @SergeCroise I have validated your proposals.

File: docs/guides/file_sharing/nfsserver.md (hunk 9 above)
Comment (SergeCroise) [reply docs/guides/file_sharing/nfsserver.md:159]: Shared resources on an NFS server are accessible through the mount point on the client.

Comment (SergeCroise) [reply docs/guides/file_sharing/nfsserver.md:159]: through a mount point
or
through the mount point

File: docs/guides/file_sharing/nfsserver.md [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+
+This command also determines whether the client workstation has authorization to mount shared resources.
+
+!!! note "note"
+
+    `showmount` sorts and hides duplicates in the results, so it's impossible to determine whether a client has made multiple mounts of the same directory or not.
+
+## Client configuration
+
+Access to shared resources on an NFS server is done by mount point on the client.
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:159]:
```suggestion
Shared resources on an NFS server are accessible through a mount point on the client.
```

File: docs/guides/file_sharing/nfsserver.md [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+* **FQDN**: client_*.rockylinux.org: allows FQDNs starting with client_ from the rockylinux.org domain;
+* `*` for everybody.
+
+Multiple clients can be specified on the same line, separated by a space.
+
+### Permissions on resources
+
+There are two types of permissions:
+
+* `ro` : read-only;
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:93]:
```suggestion
* `ro`: read-only;
```

File: docs/guides/file_sharing/nfsserver.md [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+* `rw` : modification.
```
Comment (SergeCroise) [inline docs/guides/file_sharing/nfsserver.md:94]:
```suggestion
* `rw`: read-write.
```

Comment (SergeCroise) [reply docs/guides/file_sharing/nfsserver.md:94]: https://en.wiktionary.org/wiki/read-write

**Review by SergeCroise — CHANGES_REQUESTED** [review]
@alemorvan , @sspencerwire 
please review

**Comment by sspencerwire** [comment]
@alemorvan this looks good! I've committed @SergeCroise suggestions, as they were mostly cosmetic and fixed space issues, etc. (some wording changes also were approved). I'm going to convert this to a draft and will plan on doing a little more editing tomorrow (Monday). By converting it to a draft it just means it won't be accidentally merged before the editing has been done.

**Review by wsoyinka — DISMISSED** [review]
Thank you @alemorvan  :-) !!   Beautiful contribution

**Review by SergeCroise — APPROVED** [review]

**Pull request merged**